pkg/kube/client.go

```diff
@@ -616,14 +617,35 @@ func perform(infos Result, fn ResourceActorFunc) error {
 		return ErrNoObjectsVisited
 	}
 
-	for _, info := range infos {
-		if err := fn(info); err != nil {
+	errs := make(chan error)
+	go batchPerform(infos, fn, errs)
+
+	for range infos {
+		err := <-errs
+		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func batchPerform(infos Result, fn ResourceActorFunc, errs chan<- error) {
+	var kind string
+	var wg sync.WaitGroup
+	for _, info := range infos {
+		currentKind := info.Object.GetObjectKind().GroupVersionKind().Kind
+		if kind != currentKind {
```

Comment:
won't we hit a blocker right here? At this point,

- `kind` is set to the empty string
- `currentKind` is set to the current object's kind

From what I can tell, there's no way to break out of this. We need a base condition like we did before with the `infos[0].Object.GetObjectKind().GroupVersionKind().Kind`, don't we?